### PR TITLE
fix(usbd/dcd_dwc2): Fixed epin counter assert, update buffer name for TU_LOG > 1

### PR DESCRIPTION
**Pull request by roma-jam**
## Requirements
1. When `TU_LOG` configured with value > 1 we need to provide correct name for `TU_LOG_MEM()` macros.
2. This PR contain two small fixes, which are crucial for using device configuration with fully load EP amount.
ESP32S2/S3 and ESP32P4.

## Description
1. Flushing _dcd_data.allocated_epin_count to zero while:
    - `dcd_edpt_close_all()` as we will allocate FIFO for EP0 later in `dfifo_device_init(rhport);`
    - `handle_bus_reset(uint8_t rhport)` as we will allocate FIFO for EP0 later in `dfifo_device_init(rhport);`
2. Fix the usage of counters for TU_ASSERT. When we use _dwc2_controller instead of dwc2_controller we request the ep_in_count member value from the first member, which is FS. This is important, when we run on HS PHY.
3. Changed `_usbd_ctrl_buf` -> `_ctrl_xfer.buffer`

## Limitations
N/A

## Breaking change
_No breaking changes_

## Checklist

- [x] Pull Request name has appropriate format (for example: "fix(dcd_dwc2): Resolved address selection when several phy are present")
- [ ] _Optional:_ README.md updated
- [x] CI passing

## Related issues
- Cherry-picked to the upstream: https://github.com/hathach/tinyusb/pull/2891
- Cherry-picked to the upstream: https://github.com/hathach/tinyusb/pull/2892

### CODE REVIEW

File: src/portable/synopsys/dwc2/dcd_dwc2.c
Comment (tore-espressif) [inline src/portable/synopsys/dwc2/dcd_dwc2.c]: If `_dcd_data.allocated_epin_count == _dwc2_controller->ep_in_count` next line will overflow

File: src/portable/synopsys/dwc2/dcd_dwc2.c
Comment (roma-jam) [inline src/portable/synopsys/dwc2/dcd_dwc2.c:546]: We need to `_dcd_data.allocated_epin_count = 0` here, as it will be increased after EP0 fifo allocation in `dfifo_device_init(rhport);`

File: src/portable/synopsys/dwc2/dcd_dwc2.c
Comment (roma-jam) [inline src/portable/synopsys/dwc2/dcd_dwc2.c:657]: The same here

**Review by tore-espressif — APPROVED** [review]
LGTM! We can tag a release after this is merged

**Pull request merged**
